helix-core/src/main/java/org/apache/helix/tools/ClusterSetup.java

```diff
@@ -176,6 +179,37 @@ public void close() {
     }
   }
 
+  public void addCluster(String clusterName, boolean overwritePrevious, boolean isCloudEnabled,
+      CloudConfig cloudConfig) throws HelixException {
+    if (isCloudEnabled) {
+      if (cloudConfig != null) {
+        String cloudID = cloudConfig.getCloudID();
+        String cloudProvider = cloudConfig.getCloudProvider();
+        List<String> cloudInfoSource = cloudConfig.getCloudInfoSources();
+        String cloudProcessorName = cloudConfig.getCloudInfoProcessorName();
+
+        // Validate the cloudConfig in the builder
+        // If it is invalid, do not even create a cluster
+        CloudConfig.Builder builder = new CloudConfig.Builder(clusterName);
+        builder.setCloudEnabled(true);
+        builder.setCloudID(cloudID);
+        builder.setCloudProvider(CloudProvider.valueOf(cloudProvider));
+        builder.setCloudInfoSources(cloudInfoSource);
+        builder.setCloudInfoProcessorName(cloudProcessorName);
+
+        // The builder checks the fields and generate exception with proper message.
```

Comment:
You don't need this comment here, as the "validation" is already mentioned in the previous comment.

Reply:
Done.